### PR TITLE
[inductor] Pattern to replace cumsum with arange

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100673

cc @soumith @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/fx_passes/post_grad.py [hunk 1]
```diff
@@ -114,6 +116,41 @@ def mm_plus_mm(match: Match, mat1, mat2, mat3, mat4):
     return inductor.kernel.mm_plus_mm.tuned_mm_plus_mm(mat1, mat2, mat3, mat4)
 
 
+@register_graph_pattern(
+    CallFunction(
+        aten.cumsum.default,
+        CallFunction(
+            prims.convert_element_type.default,
```
Comment (eellison) [inline torch/_inductor/fx_passes/post_grad.py:123]: presumably this is amp - could register a non amp one as well

**Review by eellison — APPROVED** [review]
cool!

File: torch/_inductor/fx_passes/post_grad.py (hunk 1 above)
Comment (jansel) [reply torch/_inductor/fx_passes/post_grad.py:123]: I don't think it is amp.  The conversion is actually float32->int64.  No idea why it is there....

**Pull request closed without merging**